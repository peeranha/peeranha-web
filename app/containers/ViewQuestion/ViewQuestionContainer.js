import React from 'react';
import PropTypes from 'prop-types';

import Box from './Box';
import Question from './Question';
import Answers from './Answers';
import AnswerQuestionForm from './AnswerQuestionForm';

const ViewQuestionContainer = props => (
  <Box>
    <Question {...props} />
    <Answers {...props} />
    <AnswerQuestionForm
      translations={props.translations}
      postAnswer={props.postAnswer}
      postAnswerLoading={props.postAnswerLoading}
    />
  </Box>
);

ViewQuestionContainer.propTypes = {
  translations: PropTypes.object,
  postAnswer: PropTypes.func,
  postAnswerLoading: PropTypes.bool,
};

export default ViewQuestionContainer;
