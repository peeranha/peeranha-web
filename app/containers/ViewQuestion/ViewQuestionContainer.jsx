import React from 'react';
import PropTypes from 'prop-types';

import AnswerForm from 'components/AnswerForm';
import Base from 'components/Base/BaseRounded';

import Question from './Question';
import Answers from './Answers';

import messages from './messages';

import {
  POST_ANSWER_BUTTON,
  ADD_ANSWER_FORM,
  QUESTION_IS_RENDERED_ID,
} from './constants';

export const ViewQuestionContainer = props => (
  <article id={QUESTION_IS_RENDERED_ID}>
    <Question {...props} />
    <Answers {...props} />

    <Base className="mt-3">
      <AnswerForm
        form={ADD_ANSWER_FORM}
        formHeader={props.translations[messages.yourAnswer.id]}
        sendButtonId={`${POST_ANSWER_BUTTON}${0}`}
        sendAnswer={props.postAnswer}
        sendAnswerLoading={props.postAnswerLoading}
        submitButtonName={props.translations[messages.postAnswerButton.id]}
      />
    </Base>
  </article>
);

ViewQuestionContainer.propTypes = {
  postAnswer: PropTypes.func,
  postAnswerLoading: PropTypes.bool,
  translations: PropTypes.object,
};

export default React.memo(ViewQuestionContainer);
