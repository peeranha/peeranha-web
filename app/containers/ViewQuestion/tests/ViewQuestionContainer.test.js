import { ViewQuestionContainer } from '../ViewQuestionContainer';

describe('ViewQuestionContainer', () => {
  it('test', () => {
    const props = {
      translations: {},
    };

    expect(ViewQuestionContainer(props)).toMatchSnapshot();
  });
});
