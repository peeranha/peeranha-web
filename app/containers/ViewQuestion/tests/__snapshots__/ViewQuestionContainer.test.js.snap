// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewQuestionContainer test 1`] = `
<article
  id="question-is-rendered"
>
  <UNDEFINED
    translations={Object {}}
  />
  <UNDEFINED
    translations={Object {}}
  />
  <Base
    className="mt-5"
  >
    <UNDEFINED
      form="app_ViewQuestion_ADD_ANSWER_FORM"
      sendButtonId="app_view_question_post_answer_button0"
    />
  </Base>
</article>
`;
